### DISCUSSION TITLE
How to restore converted files and how to disable backup during m4b conversion

### DISCUSSION THREAD

**Discussion by 0rn0lf** [advplyr/audiobookshelf#2972] [Q&A]
Hey,

I want to convert all my audiobooks from multiple mp3 files in single m4b files.
I already found the "copy" codec to speedup the whole process since i dont care about storage so much.

I'm currently running ABS as Docker container on unRaid. For the conversion, I'm using [docker-absautoconverter](https://github.com/cutzenfriend/docker-absautoconverter) from cutzenfriend which I modified to use "copy" as codec.

Now to the question, ABS always stores a backup of the converted files under _/metadata/cache/items/_
I havent found a way how to restore an audiobook from there since all books are stored with their ID and the original filepath with Author/Series/Book isnt there anymore. Is this documented somewhere? Is there a builtin fuction in the UI?

Also I havent found an option to disable the backup. The container including the metadata folder is running on my SSD pool but audiobooks are stored on my HDD array. The backup is slowing down the whole process. Also, I dont want ABS to fill up my SSD pool so I mounted the backup location to the HDD array. This works but is pretty slow and honestly useless if there isnt a way to restore certain books out of the thousand IDs to their orginal location

**Comment by advplyr** [comment]
There is no restore function built-in. If you want to restore you have to copy the audio files back into the books folder path. I'm not sure what you mean by "Author/Series/Book isn't there anymore". Abs doesn't delete your folder path so why would it not be there anymore?

There is also now way to disable backup right now. The `tone` library used to embed metadata has corrupted files before in which case you would lose the original file. We can improve that process but first `tone` is going to be removed before doing any refinement like that.

Since you are already using a third party script you can expand on that to remove the backups or move them to a new location

**Reply by 0rn0lf** [reply]
>  I'm not sure what you mean by "Author/Series/Book isn't there anymore". Abs doesn't delete your folder path so why would it not be there anymore?

No, it doesnt delete the original folder structure within the library but every audiobook is stored in its own crypic folder under /metadata/cache/items/
I was hoping for a command that can restore an audiobook automatically from the backup folder without me manually searching for the files and copying them by hand back to the original destination in the library.

> The tone library used to embed metadata has corrupted files before in which case you would lose the original file. 

So as long as i dont embed metadata, the converted files should be ok?

**Reply by nichwall** [reply]
> No, it doesnt delete the original folder structure within the library but every audiobook is stored in its own crypic folder under /metadata/cache/items/

The backup location is `/metadata/cache/items/[library itemId]` which you can get from querying the database or API.

> So as long as i dont embed metadata, the converted files should be ok?

Encoding will still have problems with some files.
